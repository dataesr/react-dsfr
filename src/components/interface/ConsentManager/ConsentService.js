import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import dataAttributes from '../../../utils/data-attributes';
import { RadioGroup, Radio } from '../Radio';

const ConsentService = forwardRef((props, ref) => {
  const {
    className,
    description,
    title,
    acceptLabel,
    refuseLabel,
    defaultConsent,
    ...remainingProps
  } = props;
  const _className = classNames('fr-consent-service ds-fr--border-bottom', className);
  return (
    <div
      className={_className}
      {...dataAttributes.getAll(remainingProps)}
      ref={ref}
    >
      <div className="fr-consent-service__radios">
        <RadioGroup
          legend={title}
          isInline
        >
          <Radio
            label={acceptLabel}
            value={`${acceptLabel.toLowerCase().replace(/\s+/g, '')}`}
            defaultChecked={defaultConsent === 'accept'}
          />
          <Radio
            label={refuseLabel}
            value={`${refuseLabel.toLowerCase().replace(/\s+/g, '')}`}
            defaultChecked={defaultConsent === 'refuse'}
          />
        </RadioGroup>
      </div>
      <p className="fr-consent-service__desc">
        {description}
      </p>
    </div>
  );
});

ConsentService.defaultProps = {
  description: '',
  className: '',
};

ConsentService.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string.isRequired,
  defaultConsent: PropTypes.oneOf(['accept', 'refuse']).isRequired,
  description: PropTypes.string,
  acceptLabel: PropTypes.string.isRequired,
  refuseLabel: PropTypes.string.isRequired,
};
export default ConsentService;
