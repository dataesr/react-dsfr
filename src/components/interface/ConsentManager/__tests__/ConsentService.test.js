import React from 'react';
import renderer from 'react-test-renderer';
import { v4 as uuidv4 } from 'uuid';
import ConsentService from '../ConsentService';

jest.mock('uuid', () => ({
  v4: jest.fn(),
}));

describe('<ConsentService />', () => {
  beforeEach(() => {
    uuidv4.mockImplementationOnce(() => 'id');
    uuidv4.mockImplementationOnce(() => 'name');
  });

  it('should render ConsentService properly', () => {
    const component = renderer
      .create(<ConsentService
        description="Description"
        title="Préférences pour tous les services."
        acceptLabel="Tout accepter"
        refuseLabel="Tout refuser"
        acceptValue="accept-all-consent"
        refuseValue="refuse-all-consent"
        consent="accept-all-consent"
      />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
