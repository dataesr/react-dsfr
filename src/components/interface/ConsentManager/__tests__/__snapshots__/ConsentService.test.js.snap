// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsentService /> should render ConsentService properly 1`] = `
<div
  className="fr-consent-service ds-fr--border-bottom"
>
  <div
    className="fr-consent-service__radios"
  >
    <div
      className="fr-form-group fr-fieldset--inline"
    >
      <fieldset
        aria-label="Préférences pour tous les services."
        className="fr-fieldset"
        disabled={false}
      >
        <legend
          className="fr-fieldset__legend fr-text--regular"
        >
          Préférences pour tous les services.
        </legend>
        <div
          className="fr-fieldset__content"
        >
          <div
            className="fr-radio-group"
          >
            <input
              defaultChecked={true}
              id="name"
              name="id"
              onChange={[Function]}
              required={false}
              type="radio"
              value="accept"
            />
            <label
              className="fr-label"
              htmlFor="name"
            >
              Tout accepter
            </label>
          </div>
          <div
            className="fr-radio-group"
          >
            <input
              name="id"
              onChange={[Function]}
              required={false}
              type="radio"
              value="refuse"
            />
            <label
              className="fr-label"
            >
              Tout refuser
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </div>
  <p
    className="fr-consent-service__desc"
  >
    Description
  </p>
</div>
`;
